### ISSUE TITLE
cmake: make catch optional

### ISSUE THREAD

**Issue by sebastianst** [Corvusoft/restbed#336]
The `CMakeLists.txt` requires `catch` via `find_package( catch REQUIRED )` and later enables testing
```
enable_testing( )
add_subdirectory( "${PROJECT_SOURCE_DIR}/test/unit" )
add_subdirectory( "${PROJECT_SOURCE_DIR}/test/feature" )
add_subdirectory( "${PROJECT_SOURCE_DIR}/test/regression" )
add_subdirectory( "${PROJECT_SOURCE_DIR}/test/integration" )
```
It would be great if package `catch` could be made optional and in case it wasn't found, testing would not be activated.

This could make Docker/CI build checkouts slimmer by allowing not to `git checkout` the `dependency/catch` submodule.

**Comment by sebastianst** [comment]
#294 I see that many others have already tried and failed...

**Comment by ben-crowhurst** [comment]
This has been resolved in the 4.7 release branch.
